### DISCUSSION TITLE
Filter on expanded nested record relations?

### DISCUSSION THREAD

**Discussion by phcreery** [pocketbase/pocketbase#1160] [Q&A] [answered]
Is it possible to filter on expanded nested record relations?

```
http://localhost:8090/api/collections/project_components/records?expand=component&filter=(expand.component.description%20~%20'res')
```

**Comment by ganigeorgiev** [comment, marked as answer]
You can filter nested relation fields but only the root records are filtered. `expand` is applied at the last step, right before returning the response, aka. on top of the filtered records.

In your example, if `component` is a `relation` field inside `project_components`, then the correct URL would be something like this:

```
http://localhost:8090/api/collections/project_components/records?expand=component&filter=(component.description%20~%20'res')
```

This will return `project_components` records that has at least one `component` relation that matches the filter. `expand` is applied on top of the returned result. If a `project_components` record has a multi-relation `components` records and one of them doesn't have description matching your criteria, the relation will be still expanded. For single relations this is not the case because the root record will be filtered and there will be nothing to expand.

**Reply by phcreery** [reply]
This worked perfectly! All I needed to do was take the `expand.` out of the filter string. Thank you.
